tpm_futurepcr/LogEvent.py

```diff
@@ -17,23 +22,55 @@
 # TPMv2: https://trustedcomputinggroup.org/wp-content/uploads/EFI-Protocol-Specification-rev13-160330final.pdf
 
 
-@dataclass(frozen=True)
-class EFIBSAData:
-    image_location: int
-    image_length: int
-    image_lt_address: int
-    device_path_vec: list[Any]
-
-
 @dataclass
-class LogEvent:
+class BaseEvent(ABC):
```

Comment:
I'm not familiar with `ABC` in Python – what's the overall purpose of BaseEvent / what's the difference between BaseEvent and GenericEvent?

Reply:
BaseEvent is a base event class that is not meant to be instantiated directly (this is achieved by inheriting from ABC, AbstractBaseClass). We can enforce methods to be overriden in children classes by defining the methods in the BaseEvent class and decorating them with the abstractmethod decorator. Currently GenericEvent and BaseEvent are mostly the same except for the show and some internal methods, but the more classes get added, if any, the more variability might be accomodated.